lightning/src/offers/offer.rs

```diff
@@ -857,9 +857,8 @@ impl OfferContents {
 		&self, amount_msats: Option<u64>, quantity: Option<u64>
 	) -> Result<(), Bolt12SemanticError> {
 		let offer_amount_msats = match self.amount {
-			None => 0,
-			Some(Amount::Bitcoin { amount_msats }) => amount_msats,
-			Some(Amount::Currency { .. }) => return Err(Bolt12SemanticError::UnsupportedCurrency),
+			Some(Amount::Bitcoin {amount_msats}) => amount_msats,
+			_ => 0,
```

Comment:
Seems we should have a unit test that given an `Offer` with a currency, we (1) successfully create an `InvoiceRequest` when not providing an amount and (2) fail creating an `InvoiceRequest` when providing an amount. We can probably update / extend `builds_invoice_request_with_amount` as needed.

Reply:
Ok, I re-checked the spec. It's fine to set the amount in an `InvoiceRequest` when the `Offer` issuer specifies a currency. The issuer will reject it if they disagree and otherwise MUST set a matching amount in the invoice.

Reply:
Hmm... wow, looks like we don't. 😕  Let's fix that in this PR or a follow-up.

Reply:
Sounds good, I'll just make the changes in this PR.